File: frontend/src/common/types/question/preview/QuestionPreviewDto.ts
import {QuestionMetadataDto} from "./QuestionMetadataDto";
import {UserMetadataDto} from "../../UserMetadataDto";
import {TagDto} from "../../TagDto";

export interface QuestionPreviewDto {
    id: number,
    title: string,
    contentShort: string,
    askedAt: string,
    tags: TagDto[],
    author: UserMetadataDto,
    metadata: QuestionMetadataDto,
}

File: backend/src/main/java/com/kk/heapoverflow/dto/question/response/QuestionPreviewDto.java
package com.kk.heapoverflow.dto.question.response;

import lombok.Data;

import java.time.LocalDateTime;

@Data
public class QuestionPreviewDto {
    private Long id;
    private String title;
    private String contentShort;
    private LocalDateTime askedAt;
    private QuestionTagDto[] tags;
    private QuestionAuthorMetadataDto author;
    private QuestionMetadataDto metadata;
}
